prometheus-node-exporter-lua: add dawn exporter

PolynomialDivision opened the pull request:
DAWN is a decentralized WiFi Controller.
https://github.com/berlin-open-wireless-lab/DAWN

The node exporter allows to gather statistics about your network.
- Infos about AP (Channel Utilization, Station Count, ...)
- Connected Clients (Signal, Capabilities)

https://github.com/berlin-open-wireless-lab/DAWN/issues/68

mweinelt commented on utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/dawn.lua:
```diff
@@ -0,0 +1,49 @@
+local ubus = require "ubus"
+
+local function scrape()
+  local metric_dawn_ap_channel_utilization = metric("dawn_ap_channel_utilization","gauge")
```
```suggestion
  local metric_dawn_ap_channel_utilization = metric("dawn_ap_channel_utilization_ratio","gauge")
```

If utilization is a percentage it should be in the range 0..1 and suffixed with ratio.

PolynomialDivision replied:
Channel utilization is a value between 0 and  255. :)

mweinelt replied:
Ok, so I believe a ratio/percentage would improve the meaning of this metric by scoping it properly.

PolynomialDivision replied:
Yep. :) I will change accordingly, just bumping dawn to latest version.

mweinelt commented on utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/dawn.lua:
```diff
@@ -0,0 +1,49 @@
+
+local function scrape()
+  local metric_dawn_ap_channel_utilization = metric("dawn_ap_channel_utilization","gauge")
+  local metric_dawn_ap_station_count = metric("dawn_ap_station_count","gauge")
```
```suggestion
  local metric_dawn_ap_station_count = metric("dawn_ap_stations_total","gauge")
```

> Note that an accumulating count has total as a suffix, in addition to the unit if applicable.

mweinelt commented on utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/dawn.lua:
```diff
@@ -0,0 +1,49 @@
+local function scrape()
+  local metric_dawn_ap_channel_utilization = metric("dawn_ap_channel_utilization","gauge")
+  local metric_dawn_ap_station_count = metric("dawn_ap_station_count","gauge")
+  local metric_dawn_station_signal = metric("dawn_station_signal","gauge")
```
```suggestion
  local metric_dawn_station_signal = metric("dawn_station_signal_dbm","gauge")
```

mweinelt commented on utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/dawn.lua:
```diff
@@ -0,0 +1,49 @@
+            bssid = ap,
+            mac = client,
+            ht_support = client_ht_support,
+           vht_support = client_vht_support,
```
Indentation is off.

mweinelt commented on utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/dawn.lua:
```diff
@@ -0,0 +1,49 @@
+        freq = ap_table['freq'],
+        ht_support = ht_support,
+        vht_support = vht_support,
+        neighbor_report = ap_table['neighbor_report'],
```
What kind of value is this? Labels should be simple strings, nothing more. The neighbor reports should probably be a separate metric.

PolynomialDivision replied:
Neighborreport is just a statis string that won't change. 802.11v is based on it. Maybe I will remove it?

mweinelt replied:
Oh, so it's more like a boolean, whether the client submits them or is capable of submitting them?

PolynomialDivision replied:
Actually there is information included that allows a client to immidiately roam to some AP (BSSID, Channel, ...)

mweinelt replied:
I guess that can change alot when a client is moving, and labels are not supposted to be short-lived.

Maybe expose those as separate metrics `dawn_station_neighbor_report_dbm` with appropriate labels?

PolynomialDivision replied:
They don't change. That is static information that APs exchange and give the client on request. :)
Like "Hey AP I want to roam, but I do not want to perform a scan, please give me roaming canidates".
Or what I do: "Hey Client here is a list of APs (as neighborreport) that you should immidiately connect to" :)

PolynomialDivision commented:
Please merge.

mweinelt commented on utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/dawn.lua:
```diff
@@ -0,0 +1,52 @@
+          vht_support = vht_support,
+          neighbor_report = ap_table['neighbor_report'],
+        }
+        metric_dawn_ap_channel_utilization_ratio(labels, ap_table['channel_utilization'])
```
Here `ap_table['channel_utilization']` is still 0..255, correct? For ratio it needs to be 0..1.

```suggestion
        metric_dawn_ap_channel_utilization_ratio(labels, ap_table['channel_utilization'] / 255)
```

neheb commented:
Is this ready to go?

PolynomialDivision commented:
> Is this ready to go?

From my side everything is fine! :) We changed everything to only export the local node and changed channel utilization to a ratio.

champtar commented:
@PolynomialDivision you need to bump the version in the Makefile

PolynomialDivision commented:
> @PolynomialDivision you need to bump the version in the Makefile

Is that correct? Do I have to bump the PKG_RELEASE or the PKG_VERSION? xD Not sure.

champtar commented:
Bump the PKG_VERSION to today and reset the PKG_RELEASE to 1

PolynomialDivision commented:
> Bump the PKG_VERSION to today and reset the PKG_RELEASE to 1

done.

champtar commented:
2020.06.26 you mean

PolynomialDivision commented:
> 2020.06.26 you mean

sry.

champtar commented:
Thanks @PolynomialDivision
